src/utils/PreparedTodos.ts

```diff
@@ -0,0 +1,28 @@
+import { Todo } from '../types/Todo';
+import { TodosFilter } from '../types/TodoFilter';
+
+export const getpreparedTodos = (
+  todos: Todo[],
+  query: string,
+  filterState: TodosFilter,
+) => {
+  let filteredTodos = [...todos];
+
+  if (query) {
+    const preparedQuery = query.toLowerCase();
+
+    filteredTodos = filteredTodos.filter(
+      ({ title }) => title.toLowerCase().includes(preparedQuery),
+    );
+  }
+
+  if (filterState === TodosFilter.Active) {
+    filteredTodos = filteredTodos.filter((todo) => !todo.completed);
+  }
+
+  if (filterState === TodosFilter.Completed) {
+    filteredTodos = filteredTodos.filter((todo) => todo.completed);
+  }
```

Comment:
`switch/case`

Reply:
Sure, I had such a solution before.
But mentor asked me to rewrite the code because,
The filter method returns a filtered copy of the array, so we don't need to create it manually.
It makes sense to use the switch design when we have 3 or more conditions, or if we want to add an error to the default part, etc., otherwise it will be shorter to use if.